### ISSUE TITLE
Different property resolution if property value is empty

### ISSUE THREAD

**Issue by ReneTrefft** [spring-projects/spring-framework#22793]
**Affects:** 5.1.6 / 4.3.23 + Spring Tools 4.2.0

```java
@Component
public class PropertiesTest {

	@Value("${key}")
	String property;

        @Value("${key}")
	String[] propertyArr;
	
}
```

If a property `key` is defined without a value in application.properties (i.e. `key=`) the variable `property` is initialized with an empty String and `propertyArr` with an empty String array.

If `key` is defined without a value in a 'Spring Boot App' Eclipse launch configuration instead ("Override Properties" feature) the start of the application fails:

```
java.lang.IllegalArgumentException: Could not resolve placeholder 'key' in value "${key}"
````

Spring should *always* resolve a property with an empty value as an empty String / String array.

**Comment by sbrannen** [comment]
It sounds to me like this is an issue with Spring Tools 4.

In light of that, I recommend you open an issue in the [Spring Tools issue tracker](https://github.com/spring-projects/sts4/issues), and I will close this issue here.

/cc @martinlippert

**Comment by martinlippert** [comment]
looks like this is indeed an issue with the boot launch support in the Spring Tools, not setting empty values from the custom properties of the launch in the same way than empty values in the properties file. Filed an issue here:

https://github.com/spring-projects/sts4/issues/275

**Comment by sbrannen** [comment]
Thanks, @martinlippert

**Comment by martinlippert** [comment]
Quick update: this is fixed now, will be released as part of the Spring Tools 4.2.1 release, coming later this week.
